### PR TITLE
Alerting: Improve tests setup

### PR DESCRIPTION
**Pull request by konrad147**
This PR introduces a document that helps writing tests in Alerting
It also adds a few helper methods for mocking data sources and APIs

**Special notes for your reviewer:**
Please take a look at the `TESTING.md` file. Let's describe there most common patters we want to see in our tests

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+We should strive to use **msw** for mocking API as often as possible.
+It gives us the closest behaviour to the real server.
+
+`public/app/features/alerting/unified/mockApi.ts` contains helper functions that speed up mocking API configuration with MSW.
+
+If you don't find a helper for an endpoint you're looking for, please add it.
+
+**Mocking using MWS forces developers to handle loading states in tests which gives us a chance to discover UI inconsistencies at very early stages**
+
+### Common API requests
```
Comment (VikaCep) [inline public/app/features/alerting/unified/TESTING.md:14]: Nice section!

File: public/app/features/alerting/unified/mocks.ts [hunk 2]
```diff
@@ -56,12 +61,15 @@ export function mockDataSource<T extends DataSourceJsonData = DataSourceJsonData
 ): DataSourceInstanceSettings<T> {
   const id = partial.id ?? nextDataSourceId++;
 
+  const uid = partial.uid ?? `mock-ds-${nextDataSourceId}`;
+
   return {
     id,
-    uid: `mock-ds-${nextDataSourceId}`,
+    uid: uid,
```
Comment (VikaCep) [inline public/app/features/alerting/unified/mocks.ts:68]:
```suggestion
    uid,
```

**Review by VikaCep — COMMENTED** [review]
Awesome work! 👏👏👏  
This helper functionality seems very useful and I think it's a great idea to write down our testing guidelines in the readme file.

File: public/app/features/alerting/unified/testSetup/datasources.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { config, setDataSourceSrv } from '@grafana/runtime';
+
+import { MockDataSourceSrv } from '../mocks';
+
+/**
+ * Sets up the data sources for the tests.
+ * Sets up both config object from grafana/runtime and the data source server
+ * @param configs data source instance settings. Use **mockDataSource** to create mock settings
+ */
+export function setupDatasources(...configs: DataSourceInstanceSettings[]) {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/testSetup/datasources.ts:11]: 👏

File: public/app/features/alerting/unified/testSetup/datasources.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const dataSources: Record<string, DataSourceInstanceSettings> = {};
+  for (const ds of configs) {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/testSetup/datasources.ts:13]: Alternatively `_.keyBy(configs)` from lodash :)

File: public/app/features/alerting/unified/TESTING.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+## Mocking API requests
+
+We should strive to use **msw** for mocking API as often as possible.
+It gives us the closest behaviour to the real server.
+
+`public/app/features/alerting/unified/mockApi.ts` contains helper functions that speed up mocking API configuration with MSW.
+
+If you don't find a helper for an endpoint you're looking for, please add it.
+
+**Mocking using MWS forces developers to handle loading states in tests which gives us a chance to discover UI inconsistencies at very early stages**
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/TESTING.md:12]:
```suggestion
**Mocking using MSW forces developers to handle loading states in tests which gives us a chance to discover UI inconsistencies at very early stages**
```

**Review by gillesdemey — APPROVED** [review]
LGTM, great job!

**Pull request merged**